persistent/Database/Persist/Class/PersistField.hs

```diff
@@ -309,21 +309,19 @@ instance PersistField UTCTime where
         let s = T.unpack t
         in
           case reads s of
-            (d, _):_ -> Right d
-            _ ->
+            [] ->
                 case parse8601 s <|> parsePretty s of
                     Nothing -> Left $ fromPersistValueParseError "UTCTime" x
                     Just x' -> Right x'
+            matches -> Right $ fst $ last matches
```

Comment:
Leafing through this PR, this was the only change that stuck out as not obvious.  
- Why switch from first match to last match?
- Could this application of `last` be made statically error-free by using `Data.List.NonEmpty.last`?

Reply:
Yeah, I mentioned this in the commit message, but the new `time` package bundled with GHC 9.2 changed the `Read UTCTime` instance to be more forgiving and provide partial matches.
```hs
-- old time
>>> reads "2018-02-27 10:49:42.12 UTC" :: [(UTCTime, String)]
[(2018-02-27 10:49:42.12 UTC,"")]

-- new time
>>> reads "2018-02-27 10:49:42.12" :: [(UTCTime, String)]
[(2018-02-27 10:49:42 UTC,"12"),(2018-02-27 10:49:42.1 UTC,"2"),(2018-02-27 10:49:42.12 UTC,"")]
```
I'll add a comment and use `NonEmpty`

Reply:
https://github.com/yesodweb/persistent/pull/1338/commits/1d4aa31a01cef2c6a40c0dc338060270e0d18171

Reply:
Excellent!